### DISCUSSION TITLE
identity server support

### DISCUSSION THREAD

**Discussion by ElderJames** [elsa-workflows/elsa-core#248] [Ideas]
Can the user system in elsa be integrated into the existing identity server system?

**Comment by sfmskywalker** [comment]
The short answer is: partially. The longer answer is: For Elsa 2.0 and the GraphQL endpoint, authentication has yet to be provided so that it's supported right out of the gate. Elsa being a set of libraries for you to configure in your own .NET Core host, you can enable Identity Server 4 yourself. However, there are some activity middlewares such as provided by the HTTP activity library that do not yet integrate with ASP.NET's authorization APIs. I will add an issue for that in order to track that functionality - thank you for raising it.

Besides that, do you have any specific scenarios in mind in terms of identity server & user support in  Elsa?

**Comment by ElderJames** [comment]
Yes, for the integration with user system, we will have the following feature request:
1. Third party login and authorization.
2. Document approval for users or roles who are assigned.
3. Multi-Tenancy. #242
